### PR TITLE
Fix data for optimized medias without next-gen versions

### PR DESCRIPTION
**Pull request by remyperona**
# Description

This PR updates the code related to storing and using the data for optimized medias without next-gen versions (depending on option selected, either WebP or AVIF).

### Technical documentation

The logic remains the same, but makes use of the new method to get the list of optimized medias without next-gen.

It also removes no longer used methods that were only accounting for WebP.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue).

# Checklists

## Feature validation

- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.

## Code style
- [x] I wrote self-explanatory code about what it does.
- [x] I wrote comments to explain why it does it.
- [x] I named variables and functions explicitely.
- [x] I did not introduce unecessary complexity.

### CODE REVIEW

File: classes/Bulk/AbstractBulk.php [hunk 1]
```diff
@@ -105,13 +105,19 @@ function_exists( 'set_time_limit' )
...
-	 * Tell if there are optimized media without WebP versions.
+	 * Tell if there are optimized media without next-gen versions.
 	 *
 	 * @since 1.9
 	 *
 	 * @return int The number of media.
 	 */
-	public function has_optimized_media_without_webp() {
-		return count( $this->get_optimized_media_ids_without_webp()['ids'] );
+	public function has_optimized_media_without_nextgen() {
```
Comment (jeawhanlee) [inline classes/Bulk/AbstractBulk.php:114]: This method seems to cater only for the enabled option, don't we need to handle any next-gen version irrespective of the enabled one?

File: classes/Stats/OptimizedMediaWithoutNextGen.php [hunk 2]
```diff
@@ -1,44 +1,37 @@
...
-	/**
-	 * Launch hooks.
-	 *
-	 * @since 1.9
-	 */
-	public function init() {
-		add_action( 'imagify_after_optimize',      [ $this, 'maybe_clear_cache_after_optimization' ], 10, 2 );
-		add_action( 'imagify_after_restore_media', [ $this, 'maybe_clear_cache_after_restoration' ], 10, 4 );
-		add_action( 'imagify_delete_media',        [ $this, 'maybe_clear_cache_on_deletion' ] );
+	public static function get_subscribed_events() {
```
Comment (jeawhanlee) [inline classes/Stats/OptimizedMediaWithoutNextGen.php:28]: Codacy complains about no doc-block for this

File: classes/Bulk/AbstractBulk.php (hunk 1 above)
Comment (remyperona) [reply classes/Bulk/AbstractBulk.php:114]: No because in that case we want to check based on the currently enabled option

**Comment by Mai-Saad** [comment]
@Tabrisrp Thanks for the PR. Is there any snippet we can use to control / overwrite next-gen status of image(s) so we can easily validate possible scenarios here?

**Comment by remyperona** [comment]
I added a method to automatically clear the counter transient when enabling/disabling `convert_to_avif`

**Comment by Mai-Saad** [comment]
@Tabrisrp Thanks for the PR. 
- The generate count is updated when activate/deactivate avif 
- Validated if avif only or webp only is enabled 
- Out of scope (not handled yet and need snippet to facilitate test) : when avif is enabled + we enable webP using filter => we can have separate GH about that
- We can have -ve missing images in the following case (can have separate GH or just add comment [here](https://github.com/wp-media/imagify-plugin/issues/760)) 

1. avif is disabled
2. upload some images (webP is created)
3. enable webP using filter
4. enable avif

note: without step 3, it will work fine

**Pull request merged**